src/ucp/wireup/wireup.c

```diff
@@ -927,13 +950,25 @@ ucs_status_t ucp_wireup_init_lanes(ucp_ep_h ep, const ucp_ep_params_t *params,
 
     /* establish connections on all underlying endpoints */
     for (lane = 0; lane < ucp_ep_num_lanes(ep); ++lane) {
+        if (ucp_ep_get_cm_lane(ep) == lane) {
+            continue;
+        }
+
         status = ucp_wireup_connect_lane(ep, params, lane, remote_address,
                                          addr_indices[lane]);
         if (status != UCS_OK) {
             return status;
         }
     }
 
+    if (ucp_ep_get_cm_lane(ep) != UCP_NULL_LANE) {
+        status = ucp_wireup_connect_lane(ep, params, ucp_ep_get_cm_lane(ep),
```

Comment:
cam unite this with line 957?
instead of addr_indices[lane] pass -1, and no harm in passing remote_address

Reply:
ordering may be important, for example, if this is a server, TL lanes must be connected before ACK